PyRDF/CallableGenerator.py

```diff
@@ -66,30 +61,22 @@ def get_callable(self):
 
         self.root_node._graph_prune()
 
-        self.actions = 0
-        self.tfs = 0
-        self.operations = []
-        self.action_node_map = {}
-        self._dfs(self.root_node)
+        def mapper(t, node=None, return_vals=[], return_nodes=[]):
```

Comment:
Can you elaborate on why this does not work with Spark? Thanks!

Reply:
@etejedor , it's because, this function (mapper) has a reference to `root`, which is a recursive structure. Since Spark pickles all these details, Pickle keeps complaining this : 
```python
PicklingError: Could not pickle object as excessively deep recursion required.
```
And also, it's return values also contain `Node` object, which is recursive and encounters the same error

Reply:
We discussed this with @bluehood yesterday. Since it seems it is the `_get_head` data member function that is causing the circular dependency (and thus the inability of pickle to serialize the graph), the `CallableGenerator` could set to `None` the `_get_head` attribute of the nodes of the graph, so that we prevent the serialization problem afterwards. Note that at this point we do not need `_get_head` anymore, since we already have a reference to the head node.
Also, once we check that this works, we should remove the `_get_head` attribute from transformation nodes, since it is only used from `Proxy`, which gets it from an `Action` node.

Reply:
@etejedor , yes. It sounds like a pretty good explanation to me. I'll reset `_get_head` to `None` 👍